### PR TITLE
docs: update community guides and resources

### PR DESCRIPTION
**Pull request by deepeshaburse**
This PR updates the "Community" section including the resources and guides.

Closed PR reference: #202 

Fixes: #189

### CODE REVIEW

File: docs/community/developer_guide.mdx [hunk 1]
```diff
@@ -66,70 +66,70 @@ Windows:
...
 To update or add new dependencies, run `go get <package name>`.
 
-## Release checklist
+## Release Checklist
 
 :::info
 
 This section needs a lot of love and automation 🙂
 
 :::
```
Comment (TerryHowe) [inline docs/community/developer_guide.mdx:86]: These should automatically number when they start with `1.`. Is that not working?

File: docs/community/developer_guide.mdx [hunk 2]
```diff
@@ -66,70 +66,70 @@ Windows:
...
-1. Download all of the artifacts uploaded to the new GitHub release locally (`*checksums.txt`, `*darwin_amd64.tar.gz`, `*linux_armv7.tar.gz`, `*linux_arm64.tar.gz`, `*linux_amd64.tar.gz`, `*windows_amd64.zip`).
-1. Verify the checksum of the file, downloaded platform should pass the check:
+6. Wait for GitHub Actions to complete successfully for both the `release-ghcr` and `release-github` pipelines.
+7. Download all of the artifacts uploaded to the new GitHub release locally (`*checksums.txt`, `*darwin_amd64.tar.gz`, `*linux_armv7.tar.gz`, `*linux_arm64.tar.gz`, `*linux_amd64.tar.gz`, `*windows_amd64.zip`).
+8. Verify the checksum of the file, downloaded platform should pass the check:
     ```sh
     shasum -c oras_${version}_checksums.txt
     ```
-1. Run version command and make sure that version number and git commit digest is what you expect it to be (same as the commit used to create the tag). Example:
+9. Run version command and make sure that version number and git commit digest is what you expect it to be (same as the commit used to create the tag). Example:
```
Comment (sabre1041) [inline docs/community/developer_guide.mdx:105]:
```suggestion
9. Run `oras version` command and make sure that version number and git commit digest is the expected value (same as the commit used to create the tag). Example:
```

File: docs/community/developer_guide.mdx [hunk 3]
```diff
@@ -66,70 +66,70 @@ Windows:
...
 
-1. Create armored GPG signatures(`.asc`) using the key in the `KEYS` file.
+10. Create armored GPG signatures(`.asc`) using the key in the `KEYS` file.
     ```sh
     for file in `ls`; do
         gpg --armor --detach-sign $file
     done
     ```
-1. Validate the signatures. Not that the `KEYS` file should be imported with `gpg --import KEYS`. Run some form of the following (adapted from Linux project):
+11. Validate the signatures. Not that the `KEYS` file should be imported with `gpg --import KEYS`. Run some form of the following (adapted from Linux project):
```
Comment (sabre1041) [inline docs/community/developer_guide.mdx:118]:
```suggestion
11. Validate the signatures. Note that the `KEYS` file should be imported with `gpg --import KEYS`. Run some form of the following (adapted from Linux project):
```

File: docs/community/developer_guide.mdx [hunk 4]
```diff
@@ -66,70 +66,70 @@ Windows:
...
     ```sh
     for file in `ls *.asc`; do
         gpg --verify $file
     done
     ```
-1. Click "Edit release" button on the release, and add the `.asc` files created in the previous step. Edit the release description for change logs and also add a note indicating your GPG key used to sign the artifacts. Example (replace with your fingerprint etc.):
+12. Click "Edit release" button on the release, and add the `.asc` files created in the previous step. Edit the release description for change logs and also add a note indicating your GPG key used to sign the artifacts. Example (replace with your fingerprint etc.):
```
Comment (sabre1041) [inline docs/community/developer_guide.mdx:124]:
```suggestion
12. Click the "Edit release" button on the release, and add the `.asc` files created in the previous step. Edit the release description for change logs and also add a note indicating your GPG key used to sign the artifacts. Example (replace with your fingerprint etc.):
```

File: docs/community/developer_guide.mdx [hunk 5]
```diff
@@ -66,70 +66,70 @@ Windows:
...
     ## Notes
 
     This release was signed with `BE6F A8DD A48D 4C23 0091 A0A9 276D 8A72 4CE1 C704` (@qweeah's GPG key) which can be found [here](https://github.com/qweeah.gpg).
     ```
-1. Click "Publish Release" button to save. Double-check that the release contains a corresponding `.asc` file for each release artifact.
-1. Consume beverage of choice.. you're done! Thanks for moving the project forward.
-1. Oh yea, tell people about it in `#oras`
+13. Click "Publish Release" button to save. Double-check that the release contains a corresponding `.asc` file for each release artifact.
+14. Consume beverage of choice.. you're done! Thanks for moving the project forward.
+15. Oh yea, tell people about it in `#oras`.
```
Comment (sabre1041) [inline docs/community/developer_guide.mdx:132]:
```suggestion
15. Oh yea, tell people about it in the `#oras` slack channel.
```

**Review by sabre1041 — CHANGES_REQUESTED** [review]

File: docs/community/developer_guide.mdx (hunk 1 above)
Comment (deepeshaburse) [reply docs/community/developer_guide.mdx:86]: It is, I thought that was done by accident in the markdown though. I'll revert these changes in the next commit.

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Review by sabre1041 — APPROVED** [review]

**Pull request merged**